core/src/main/java/org/apache/metamodel/query/FromClause.java

```diff
@@ -75,7 +75,13 @@ public FromItem getItemByReference(String reference) {
         return null;
     }
 
-    private FromItem getItemByReference(FromItem item, String reference) {
+    private FromItem getItemByReference(final FromItem item, final String reference) {
+        if (reference.equals(item.toStringNoAlias(false))) {
+            return item;
+        }
+        if (reference.equals(item.toStringNoAlias(true))) {
+            return item;
+        }
         final String alias = item.getAlias();
```

Comment:
Wouldn't it be better to write just one if instead of two? I'd rewrite it to:

if (reference.equals(item.toStringNoAlias(false)) || reference.equals(item.toStringNoAlias(true))) {
            return item;
}